src/main/java/org/broadinstitute/hellbender/dev/pipelines/bqsr/ReadsFilter.java

```diff
@@ -0,0 +1,35 @@
+package org.broadinstitute.hellbender.dev.pipelines.bqsr;
+
+import com.google.api.services.genomics.model.Read;
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.genomics.gatk.common.GenomicsConverter;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+
+/**
+ * A simple filter for reads. Pulled into its own class due to Java limitations.
+ */
+public final class ReadsFilter extends DoFn<Read, Read>   {
+
+    private final ReadFilter readFilter;
+    private final SAMFileHeader header;
+
+    /**
+     * ReadsFilter will use the given header to interpret each read, and then let the Read through
+     * if the passed filter accepts it.
+     */
+    public ReadsFilter(final ReadFilter filterToApply, final SAMFileHeader header) {
+        this.readFilter = filterToApply;
+        this.header = header;
+    }
+
+    @Override
+    public void processElement(ProcessContext c) throws Exception {
+        Read r = c.element();
+        final SAMRecord sr = GenomicsConverter.makeSAMRecord(r, header);
```

Comment:
Add a `TODO` comment here to remove this conversion once we switch to a `Read` interface (so that I don't forget to change it :)